src/connection_string.ts

```diff
@@ -955,16 +957,14 @@ export const OPTIONS = {
           readPreference: { ...options.readPreference, ...value },
           ...value
         } as any);
-      }
-      if (isRecord(value, ['mode'] as const)) {
+      } else if (isRecord(value, ['mode'] as const)) {
```

Comment:
why the additions of `else` here all the returns are already definitely typed, so this seems like unnecessary logic, minus the last throw; if anything, this syntax implies that the block above might *not* properly return

Reply:
Good point, there was no need for else(s), reverted, kept the throws.